Non native arch builds recommendations (#13037)
kyma-project/test-infra: cmd/image-builder/README.md

  enhanced workflow experience** comparable to the GitHub workflow approach.
- **Build Report**: A dedicated flag (`--build-report-path`) has been added to the `image-builder` binary, enabling it to write a build
  report generated by the ADO pipeline to a specified file. This functionality is intended specifically for use in Jenkins.

> ## Cross-Compiling and Caching for Non Native Architecture Builds

Reviewer: ```suggestion
## Cross-Compiling and Caching for Non-Native Architecture Builds
```

Image Builder uses builder agents with `linux/amd64` native architecture.

> When building images for multiple architectures or building image for non native architecture,

Reviewer: ```suggestion
When building images for multiple architectures or building an image for a non-native architecture,
```

consider enabling cross-compilation to significantly reduce build times.

> Testing has shown that cross-compilation can speed up the build process by **10x**.
> Reducing build times from 12 minutes to less than 2 minutes in our tests scenario with rather small golang codebase.

Reviewer: ```suggestion
Testing has shown that cross-compilation can speed up the build process by **10x**, reducing build times from 12 minutes to less than 2 minutes in our test scenario with a rather small golang codebase.
```

### Key Recommendations

> - **Cross-Compilation**: If you are building, builder non native architecture images, implement cross-compilation in your Dockerfile, use

Reviewer: ```suggestion
- **Cross-Compilation**: If you are building non-native architecture images, implement cross-compilation in your Dockerfile, use
```

Reviewer: The word 'builder' doesn't seem to fit this context. I removed it, but it's possible that part of this sentence that complements the builder idea is missing.

  the [Faster Multi-Platform Builds: Dockerfile Cross-Compilation Guide](https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide/)
  as a reference.

> - **Bind Mounts**: You can use bind mounts for `RUN` command in Dockerfiles to avoid copying source code for compilation.

Reviewer: ```suggestion
- **Bind Mounts**: To avoid copying source code for compilation, use bind mounts for the `RUN` command in Dockerfiles.
```

>   Although the speed gain was minimal in our tests. We achieved speed up of less than ~5 seconds.

Reviewer: ```suggestion
  However, the speed gain was minimal in our tests: We achieved a speedup of less than ~5 seconds.
```

- **Cache Mounts for Go Compiler**:
  Rely on a **cache backed by a remote repository**, because a new agent is allocated for each pipeline execution, making mount-type caching
  ineffective.

>   Use mounts a cache type for Go package downloads, and binary compilation cache did not bring speed increase during tests.

Reviewer: ```suggestion
  Use mounts a cache type for Go package downloads. The binary compilation cache did not increase speed during tests.
```

Reviewer: "Use mounts a cache type" sounds very wrong to me, but maybe it's a fixed phrase. Is it? If not, please rephrase.

Author: This is to say to use a `mounts` that has type `cache`.
